### PR TITLE
Feature chinedu profile update

### PR DESCRIPTION
**Pull request by ChineduOguejiofor**
#### What does this PR achieve?
- User Can update profile
- A vendor can update profile
- Added more files to the signup
- Formats created date to human-readable dates

#### Relevant Trello Tickets

- [User Vendor Profile Backend](https://trello.com/c/hNNVAjel/31-update-profile-api-user-vendor)

### CODE REVIEW

File: functions/middleware/auth.js [hunk 1]
```diff
@@ -57,6 +57,28 @@ exports.checkIfUser = (req, res, next) => {
...
+            if (userInfo.vendor === true) {
+                req.authId = userInfo.uid;
+                return next();
+            }
+
+            throw new Error('unauthorized');
+        } catch (e) {
+            return res
+                .status(UNAUTHORIZED)
+                .send({ error: 'You are not authorized to make this request' });
```
Comment (enyeInc) [inline functions/middleware/auth.js]: use the messages from `http-status-code`

File: functions/controllers/profile.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        await db.doc(`addresses/${userID}`).set({
+            ...addresses,
+        });
+        return res
+            .status(OK)
+            .json({ status: 'success', data: { info: 'Update successful' } });
+    } catch (error) {
+        return res
+            .status(INTERNAL_SERVER_ERROR)
+            .json({ status: 'error', msg: 'There was an error' });
```
Comment (enyeInc) [inline functions/controllers/profile.js]: use the messages from http-status-code. and its message not `msg`

File: functions/controllers/auth.js [hunk 3]
```diff
@@ -20,8 +13,12 @@ const sendEmailhelper = async email => {
...
 };
 
-exports.hello = (req, res) => {
-    res.send('You have hit the users route');
+exports.helloUser = (req, res) => {
+    res.json({ msg: 'Hello world from protected user route' });
+};
+
+exports.helloVendor = (req, res) => {
+    res.json({ msg: 'Hi from protected vendor route' });
```
Comment (enyeInc) [inline functions/controllers/auth.js]: message not `msg`

File: functions/middleware/auth.js (hunk 1 above)
Comment (ChineduOguejiofor) [reply functions/middleware/auth.js]: updated
Thanks

File: functions/controllers/auth.js (hunk 3 above)
Comment (ChineduOguejiofor) [reply functions/controllers/auth.js]: removed
Thanks

**Comment by ChineduOguejiofor** [comment]
> Good Job @ChineduOguejiofor . Please fix conflict.



> Good Job @ChineduOguejiofor . Please fix conflict.
conflicts have been fixed

File: functions/controllers/profile.js (hunk 2 above)
Comment (ChineduOguejiofor) [reply functions/controllers/profile.js]: updated
thanks

**Pull request merged**